back/engines/commercial/multi_tenancy/app/services/multi_tenancy/templates/tenant_serializer.rb

```diff
@@ -172,14 +165,29 @@ def sort_by_references(models)
       #   }
       #
       # @return [Hash<Class, Array<Class>>]
-      def extract_referential_dependencies(models)
-        models.transform_values do |records|
+      def extract_dependencies_graph(models)
+        graph = models.transform_values do |records|
           records.flat_map do |_id, attributes|
             attributes.values.filter do |value|
               value.is_a?(Serializers::Core::Ref) && value.id
             end.map(&:klass)
           end.uniq
         end
+
+        # User depends on CustomField because of the custom field values. We have to add
+        # this dependency manually because the custom field values are stored as JSON in
+        # the database, so serialized users don't hold references to CustomField.
+        if graph.key?(User) && graph.key?(CustomField)
+          graph[User] << CustomField
+        end
```

Comment:
This looks unrelated to the main thrust of this PR.

Were we not including some `custom_fields` related to `users` when copying (anonymized) users via our templating system, prior to this change?

Reply:
@jinjagit This code was moved, so it was already there. I'm not sure if it needs to stay or can be removed, but I didn't want to take the risk removing it.